src/test/java/lambda/part3/exercise/ReachIterableTest.java

```diff
@@ -46,32 +46,41 @@
             );
         }
 
-        default <R> ReachIterable<R> flatMap(final Function<T, List<R>> function) {
-            final List<R> results = new ArrayList<>();
-            while (this.forNext(element -> results.addAll(function.apply(element))));
-            return ReachIterable.from(results);
+        default <R> ReachIterable<R> flatMap(final Function<T, List<R>> mapper) {
+            final AtomicReference<ReachIterable<R>> inner = new AtomicReference<>(
```

Comment:
I thought Atomic* wrapper is easier to read and work with than writing to array, isn't it? I understand that volatile can be expansive, but does it matter here?
I'd prefer to use simple mutable value-object for this purpose, but can't make private inner class for in interface.